lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -859,7 +863,17 @@ private function Annotation()
             );
         }
 
-        $values = $this->MethodCall();
+        $defaultProperty = 'value';
```

Comment:
Instead of this code block and passing down `$defaultProperty` could you instead handle this logic in the original line 944 and following?

```php
 foreach ($values as $property => $value) {
    if ($property === 'value') {
        $property = self::$annotationMetadata[$name]['default_property'];
    }
```
That would be less invasive and keep the logic at fewer places. I am not 100% sure it could work this way though, just a rough idea.

Reply:
Hey @beberlei! Thx for the review. I didn't want to pass the `defaultProperty` down. The problem is, without it, we don't know with the return of `values()`, if `value` as been set because it's the default property... or because we have an argument named
`value`. So it would not work if for example the constructor looks like this:
```
public function __construct(string $name, int $value){}
```

Reply:
But isnt that already a problem? Or just for the new named constructors?

Reply:
I gave a bunch of examples here : https://github.com/doctrine/annotations/issues/396.
The fact that we can't know if `value` has been set because it's the default property or because we have an argument with the same name, is, indeed an "old problem".
But we didn't want to change the current behaviour to avoid BC breaks, so we decide to handle it properly only with the new `@NamedArgumentsConstructor` annotation as it is not published yet.

Reply:
The Problem is all methods with large letter starting are named after the rules in the grammer. None of them accept arguments, because they already operate on state of the class instead. Passing arguments now will create a huge inconsistency in the API.

What you could try instead is change the return value of `Values` to be an associative array with: `['named_arguments' => [], 'positional_arguments' => []]` and then modify `Annotation` to act accordingly.

Reply:
@beberlei Here is my first attempt trying what you suggested.
I was also wondering why this is allowed: `@Name(@Name, @Name)`, but this is not `@Name("1", "2")`. The second case is not allowed because of this line: https://github.com/doctrine/annotations/blob/b81688bc95a16b76b2849bb67004d8f34ddfffd7/lib/Doctrine/Common/Annotations/DocParser.php#L1037
In our case, it means that we can only use one positional argument. It would make sense for us to allow multiple positional arguments as it's also how attributes work. `#[Name("1", "2")]` would work for example as long as the related attribute constructor accept at least two arguments. 
Removing the above line don't break the tests and allow multiple positional arguments. 
What do you think about this?

Reply:
> In our case, it means that we can only use one positional argument. It would make sense for us to allow multiple positional arguments as it's also how attributes work.

Very well, but Doctrine Annotations only supports one positional argument: the default property. I would consider it out of scope of this PR to introduce the ability to have multiple ones.

Besides, a developer using an annotation does not know whether it has been implemented with a named constructor or not (and they should not need to know, imho). It would be kind of odd if multiple positional arguments only worked in the named argument constructor case.

Reply:
Anyway, I'm a bit late to the party and I see that the discussion has already moved on. I'm fine with allowing multiple positional arguments. Thank you for working on this feature. 😃

Reply:
@derrabus : Actually, Annotations already support multiple positional arguments. It's just that without `@NamedArgumentConstructor` all positional values will be populated as an array into `value`. This `@Name(@Name, @Name)` already work. So in a developer point of view, it doesn't really change anything.